src/algebra/algebra/basic.lean

```diff
@@ -999,7 +999,8 @@ symm_bijective.injective $ ext $ λ x, rfl
 
 @[simp] theorem refl_symm : (alg_equiv.refl : A₁ ≃ₐ[R] A₁).symm = alg_equiv.refl := rfl
 
-@[simp] lemma to_ring_equiv_symm (f : A₁ ≃ₐ[R] A₁) : (f : A₁ ≃+* A₁).symm = f.symm := rfl
+--this should be a simp lemma but causes a lint timeout
```

Comment:
Have you considered using the new `alg_equiv_class`?

Reply:
Good point, I'll have a look at that!

Reply:
@riccardobrasca I had a go at it, would this be the correct statement? 
`lemma to_ring_equiv_symm {F : Type*} [alg_equiv_class F R A₁ A₂] (f : F) : ((f : A₁ ≃ₐ[R] A₂) : A₁ ≃+* A₁).symm = (f : A₁ ≃ₐ[R] A₂).symm := rfl`

Reply:
The statement doesn't compile so I'm not sure it's correct, although I think some coercions might be missing since Lean threw me an error when I wrote `(f : A₁ ≃ₐ[R] A₂)`.

Reply:
I think the idea of `alg_equiv_class` is to make this kind of lemmas useless, but I am indeed not sure it can be used here. @Vierkantor what do you think (maybe you also know why `simp` is causing a timeout)?

Reply:
Would it be reasonable to merge this PR so we can then merge #15000  and then fix the lemma in a subsequent PR, or would it be better to do the fix now?